ndk/src/signers/nip46/backend/index.ts

```diff
@@ -82,14 +87,21 @@ export class NDKNip46Backend {
      */
     public constructor(
         ndk: NDK,
-        privateKeyOrSigner: string | NDKSigner,
+        privateKeyOrSigner: string | Uint8Array | NDKSigner,
         permitCallback: Nip46PermitCallback
     ) {
         this.ndk = ndk;
-        this.signer =
-            typeof privateKeyOrSigner === "string"
-                ? new NDKPrivateKeySigner(privateKeyOrSigner)
-                : privateKeyOrSigner;
+        
+        if (privateKeyOrSigner instanceof Uint8Array) {
```

Comment:
We have to handle both string and `Uint8Array` private keys as inputs for backwards compatability